virtual_rainforest/main.py

```diff
@@ -0,0 +1,126 @@
+"""Defines the function used to run a full simulation of the model.
+
+As well as setting up the function to run the overall virtual rainforest simulation,
+this script also defines the command line entry points for the model.
+"""
+
+from copy import deepcopy
+from typing import Any, Optional, Type, Union
+
+from virtual_rainforest.core.config import validate_config
+from virtual_rainforest.core.logger import LOGGER, log_and_raise
+from virtual_rainforest.core.model import MODEL_REGISTRY, BaseModel, InitialisationError
+
+
+def select_models(model_list: list[str]) -> Optional[list[Type[BaseModel]]]:
+    """Select the models to be run for a specific virtual rainforest simulation.
+
+    This function looks for models from a list of models, if these models can all be
+    found in the registry then they are returned. Otherwise an error is logged, which
+    should be handled appropriately downstream.
+
+    Args:
+        model_list: A list of models to select
+
+    Returns:
+        modules: A set of models to be configured
+    """
+
+    # Remove "core" from model list as it is not a model
+    if "core" in model_list:
+        model_list.remove("core")
+
+    LOGGER.info(f"Attempting to configure the following models: {model_list}")
+
+    # Make list of missing models, and return an error if necessary
+    miss_model = [model for model in model_list if model not in MODEL_REGISTRY.keys()]
+    if miss_model != []:
+        LOGGER.error(
+            f"The following models cannot be configured as they are not found in the "
+            f"registry: {miss_model}"
+        )
+        return None
```

Comment:
I don't immediately see the value in having `select_models` return None here, rather than than simply raising the error, which is currently deferred to `vr_run`? Just fail and have a simpler return value (no Optional).

Reply:
Yeah I agree, easier to understand the higher level function if a `try: except` block is used instead of `if is None:` as well

Reply:
Here, you don't even need the higher level `try` since you're `log_and_raise`-ing. You can just let select_models fail and do the `log_and_raise` there.